lib/github_advisory_sync.rb

```diff
@@ -70,88 +59,102 @@ def github_api(adapter = :net_http)
     # error checking and how queries and requests are formed
     def github_graphql_query(graphql_query_name, graphql_variables = {})
       graphql_query_str = GraphQLQueries.const_get graphql_query_name
-      graphql_body = JSON.generate query: graphql_query_str,
-                                   variables: graphql_variables
+      graphql_body = JSON.generate(query: graphql_query_str, variables: graphql_variables)
+
       puts "Executing GraphQL request: #{graphql_query_name}. Request variables:\n#{graphql_variables.to_yaml}\n"
+
       faraday_response = github_api.post do |req|
         req.url GITHUB_API_URL
         req.body = graphql_body
       end
+
       puts "Got response code: #{faraday_response.status}"
+
       if faraday_response.status != 200
-        raise(GitHubGraphQLAPIError, "GitHub GraphQL request to #{faraday_response.env.url} failed: #{faraday_response.body}")
+        raise(
+          GitHubGraphQLAPIError,
+          "GitHub GraphQL request to #{faraday_response.env.url} failed: #{faraday_response.body}"
+        )
       end
+
       body_obj = JSON.parse faraday_response.body
-      if body_obj["errors"]
-        raise(GitHubGraphQLAPIError, body_obj["errors"].map { |e| e["message"] }.join(", "))
-      end
+
+      raise(GitHubGraphQLAPIError, body_obj["errors"].map { |e| e["message"] }.join(", ")) if body_obj["errors"]
+
       body_obj
     end
 
-    def retrieve_all_github_advisories(max_pages = 1000, page_size = 100)
-      all_advisories = []
+    def all_rubygem_advisories
+      advisories = {}
+
+      retrieve_all_rubygem_vulnerabilities.each do |vulnerability|
+        advisory = GitHubAdvisory.new(vulnerability["advisory"])
+
+        next if advisory.withdrawn?
+
+        advisories[advisory.primary_id] = advisory unless advisories.key? advisory.primary_id
```

Comment:
i believe this will do just fine too 😄 

```suggestion
        advisories[advisory.primary_id] ||= advisory
```